src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java

```diff
@@ -247,4 +235,40 @@ public int compare(StepDescriptor o1, StepDescriptor o2) {
         private static final long serialVersionUID = 1L;
     }
 
+    @Restricted(DoNotUse.class)
+    @Extension public static class PerJobAdder extends TransientActionFactory<Job> {
+
+        @Override public Class<Job> type() {
+            return Job.class;
+        }
+
+        @Override public Collection<? extends Action> createFor(Job target) {
+            // TODO probably want an API for FlowExecutionContainer or something
```

Comment:
Would make things worse for two reasons:

* `LocalAction` has to extend `Snippetizer`, which would force `workflow-job` to get a dependency on `workflow-cps`.
* Downstream PRs show callers which currently have a `workflow-cps` dependency anyway which would need to add a `workflow-job` dependency.